### PR TITLE
feat: Add walkthrough

### PR DESCRIPTION
**Pull request by Hellebore**
First draft of walkthrough

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,125 +1,221 @@
...
+						"title": "Add custom rules",
+						"description": "You can add your own custom rules to Sourcery!",
+						"media": {
+							"markdown": "walkthrough/custom_rules_tutorial.md"
+						}
+					},
+					{
+						"id": "sourcery.enterToken",
+						"title": "Enter your token",
+						"description": "Sign up for a Sourcery token to get a free trial of Sourcery Pro for 30 days\n[Sign up for token](https://sourcery.ai/signup)\n[Enter token](command:toSide:workbench.action.openSettings?\"sourcery\")",
```
Comment (brendanator) [inline package.json:82]: This change filters it down to sourcery settings only
```suggestion
						"description": "Sign up for a Sourcery token to get a free trial of Sourcery Pro for 30 days\n[Sign up for token](https://sourcery.ai/signup)\n[Enter token](command:toSide:workbench.action.openSettings?\"sourcery:\")",
```

File: package.json [hunk 2]
```diff
@@ -1,125 +1,221 @@
...
+						"media": {
+							"markdown": "walkthrough/enter_token.md"
+						},
+						"completionEvents": [
+							"onSettingChanged:sourcery.token"
+						]
+					},
+					{
+						"id": "sourcery.scanRefactorings",
+						"title": "Do a scan for suggestions",
```
Comment (brendanator) [inline package.json:92]:
```suggestion
						"title": "Scan a directory for suggestions",
```

File: package.json [hunk 3]
```diff
@@ -1,125 +1,221 @@
...
+						"description": "Sign up for a Sourcery token to get a free trial of Sourcery Pro for 30 days\n[Sign up for token](https://sourcery.ai/signup)\n[Enter token](command:toSide:workbench.action.openSettings?\"sourcery\")",
+						"media": {
+							"markdown": "walkthrough/enter_token.md"
+						},
+						"completionEvents": [
+							"onSettingChanged:sourcery.token"
+						]
+					},
+					{
+						"id": "sourcery.scanRefactorings",
```
Comment (brendanator) [inline package.json:91]: I'm not sure if this is worth including here.

An alternative final step is to tell people to browse our docs and include links to e.g:
- Docs main page
- Scan for refactorings
- Detect clones
- Config

File: package.json [hunk 4]
```diff
@@ -1,125 +1,221 @@
...
+				"title": "Get started with Sourcery",
+				"description": "Your first steps to try out the powerful tools and features that the Sourcery extension has to offer!",
+				"when": "workspacePlatform != webworker",
+				"steps": [
+					{
+						"id": "sourcery.openWelcomeFile",
+						"title": "Get your first recommendation",
+						"description": "Get started by opening our tutorial \n[Open tutorial](command:toSide:sourcery.welcome.open)\n Put the cursor over the underlined code to see the refactoring suggestion. \n You can see all Sourcery's recommendations in the [Problems Pane](command:workbench.actions.view.problems)",
+						"media": {
+							"markdown": "walkthrough/view_welcome_file.md"
```
Comment (brendanator) [inline package.json:41]: Please add something to this - you could probably move some of the steps out of custom rules, and replace the screenshots

Comment (Hellebore) [reply package.json:41]: Will address this in follow-up PR

File: package.json (hunk 3 above)
Comment (Hellebore) [reply package.json:91]: Fleshing this out in follow-up PR

**Review by angrycaptain19 — APPROVED** [review]
Looking really good. I'm not 100% sure we want the final part of the walkthrough to be "Sourcery Pro free trial" - but I don't have a better idea, so let's go with it and see how people interact.

**Pull request merged**